### PR TITLE
[receiver/redis] include server.address and server.port resource attributes

### PR DESCRIPTION
**Pull request by sigilioso**
**Description:** <Describe what has changed.>

Two new resource attributes have been added: `server.address` and `server.port`. This is change aligned with [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.25.0/specification/semantic-conventions.md?plain=1#L30) and allows users identifying a particular redis server.

**Link to tracking Issue:** #22044

**Testing:** <Describe what testing was performed and which tests were added.>

```
❯ go test -race -timeout 300s -parallel 4 --tags="",integration --count 1 ./...
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/redisreceiver        6.179s
ok      github.com/open-telemetry/opentelemetry-collector-contrib/receiver/redisreceiver/internal/metadata      1.464s
```

* Added: new test to check error when the configuration endpoint is missing or invalid
* Updated: integration test to check new resource attributes

**Documentation:** <Describe the documentation added.>
New attribute documentation was generated using `mdatagen`

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

**Comment by sigilioso** [comment]
Hi @dmitryax, @hughesjj, I was wondering if you could provide some feedback. Do you consider that identifying redis server by including those attributes is useful?

**Comment by ceevaaa** [comment]
Hey guys,

I am facing a similar issue while collecting metrics from five different Redis instances.
There is no way to identify which metrics came from which instance. I am sure adding server and port in resource will significantly help in uniquely identifying the Redis instance.

My question is, when is this PR going to get merged, and when is it expected to be released ?

Big thanks for all the hard-work <3.

File: receiver/redisreceiver/redis_scraper.go [hunk 1]
```diff
@@ -48,11 +50,16 @@ func newRedisScraper(cfg *Config, settings receiver.CreateSettings) (scraperhelp
 }
 
 func newRedisScraperWithClient(client client, settings receiver.CreateSettings, cfg *Config) (scraperhelper.Scraper, error) {
+	configInfo, err := newConfigInfo(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("invalid config %w", err)
```
Comment (atoulme) [inline receiver/redisreceiver/redis_scraper.go:55]:
```suggestion
		return nil, err
```

File: receiver/redisreceiver/metadata.yaml [hunk 2]
```diff
@@ -13,6 +13,14 @@ resource_attributes:
...
     enabled: true
     type: string
+  server.address:
+    description: Redis server's address
+    enabled: true
+    type: string
+  server.port:
+    description: Redis server's port
+    enabled: true
+    type: string
```
Comment (atoulme) [inline receiver/redisreceiver/metadata.yaml:23]: Is the port a string?

File: receiver/redisreceiver/metadata.yaml [hunk 3]
```diff
@@ -13,6 +13,14 @@ resource_attributes:
     description: Redis server's version.
     enabled: true
     type: string
+  server.address:
+    description: Redis server's address
+    enabled: true
+    type: string
+  server.port:
+    description: Redis server's port
+    enabled: true
```
[the remark this answers is not in the document]
Comment (sigilioso) [reply receiver/redisreceiver/metadata.yaml:22]: As I see it, these resource attributes should be enable by default. Although, If we agreed to do so, I would need to change the changelog entry to make it breaking. Do you recommend disabling the attributes by default or making the breaking change?

File: receiver/redisreceiver/metadata.yaml (hunk 2 above)
Comment (sigilioso) [reply receiver/redisreceiver/metadata.yaml:23]: We could parse it to get the corresponding number, but it is a string when [it is taken from the address](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/26707/files#diff-6d24e7ef9f7296d9beed1c934158a4b6b1ef055abe3ea78ea78a0460a6e7f7f8R48)

File: receiver/redisreceiver/metadata.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (atoulme) [reply receiver/redisreceiver/metadata.yaml:22]: You always do this in 2 separate PRs so folks can try it out for a bit.

[the remark this answers is not in the document]
Comment (sigilioso) [reply receiver/redisreceiver/metadata.yaml]: Attributes disabled here: [1078ee7](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/26707/commits/1078ee72a441766ecda949c8220564dedd88ac7c)

File: receiver/redisreceiver/metadata.yaml (hunk 2 above)
Comment (atoulme) [reply receiver/redisreceiver/metadata.yaml:23]: it's fine, let's leave it as is

**Review by atoulme — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
pinging @dmitryax and @hughesjj as code owners

**Review by mx-psi — APPROVED** [review]

**Pull request merged**